modules/build/src/main/scala/scala/build/CLangInstaller.scala

```diff
@@ -0,0 +1,111 @@
+package scala.build
+
+import coursier.cache.FileCache
+import coursier.util.{Artifact, Task}
+import os.Path
+
+import java.net.{HttpURLConnection, URL}
+import java.util.Locale
+
+import scala.build.EitherCps.either
+import scala.build.errors.BuildException
+import scala.build.internal.Runner
+import scala.util.{Properties, Try}
+
+final class CLangInstallException(message: String)
+    extends BuildException(s"Failed to Install CLang: $message")
+
+object CLangInstaller {
+  val mambaApiUrl = "https://micromamba.snakepit.net/api/micromamba"
+
+  def directories = scala.build.Directories.default()
+  /*
+  Installer based on micromamba attempts to install mamba and llvm based on that
+  Following https://gist.github.com/wolfv/fe1ea521979973ab1d016d95a589dcde
+
+  micromamba is available for 6 architectures: https://anaconda.org/search?q=micromamba
+   */
+
+  def install(logger: Logger) =
+    platform
+      .flatMap(resolveMicroMambaBinaryUrl(_))
+      .flatMap(binaryUrl => fetchFile(binaryUrl, logger))
+      .flatMap(archive => doInstall(archive, logger))
+
+  def platform = {
+    val os = Properties.osName match {
+      case os if os.startsWith("Windows")  => "win"
+      case os if os.startsWith("Mac OS X") => "osx"
+      case os if os.startsWith("Linux")    => "linux"
+    }
+    // parsing os.arch example =>  https://github.com/trustin/os-maven-plugin/blob/master/src/main/java/kr/motd/maven/os/Detector.java
+    val arch = sys.props("os.arch").toLowerCase(Locale.ROOT)
+    (os, arch) match {
+      case ("win", "x86_64")    => Right("win-64")
+      case ("osx", "x86_64")    => Right("osx-64")
+      case ("osx", "arm64")     => Right("osx-arm64")
+      case ("linux", "x86_64")  => Right("linux-64")
+      case ("linux", "aarch64") => Right("linux-aarch64")
+      case ("linux", "ppc64le") => Right("linux-ppc64le")
+      case _ => Left(new CLangInstallException("Unsupported architecture/system combination."))
+    }
+  }
+
+  def resolveMicroMambaBinaryUrl(platform: String) = Try {
+    val con: HttpURLConnection = new URL(s"$mambaApiUrl/$platform/latest")
+      .openConnection()
+      .asInstanceOf[HttpURLConnection]
+    con.setInstanceFollowRedirects(false)
+    con.connect()
+    if (con.getResponseCode != 307)
+      new CLangInstallException("Failed to resolve micromamba archive url")
+    s"https:${con.getHeaderField("Location")}"
+  }.toEither
+
+  def fetchFile(url: String, logger: Logger) = either {
+    val cache = FileCache().withLogger(logger.coursierLogger)
+    cache.logger.use {
+      cache.file(Artifact(url).withChanging(true)).run.flatMap {
+        case Left(e)  => Task.fail(new Exception(e))
+        case Right(f) => Task.point(os.Path(f, os.pwd))
+      }.unsafeRun()(cache.ec)
+    }
+  }.left.map(_ => new CLangInstallException("Failed to fetch Mamba binary."))
```

Comment:
This discard the original exception. Better change `CLangInstallException` so that it accepts a cause:
```suggestion
  }.left.map(e => new CLangInstallException("Failed to fetch Mamba binary", e))
```